### PR TITLE
Extend provider for transfer and balance

### PR DESCRIPTION
**Pull request by b0xtch**
Extend provider interface to support `getBalance` and `getTransfer`

### CODE REVIEW

File: src/Provider.ts [hunk 1]
```diff
@@ -26,6 +41,8 @@ export interface ProviderInterface {
   isConnected(): Promise<boolean>;
   principal: Principal;
   withDankProxy: WithDankProxy;
+  requestBalance(accountId: number): Promise<bigint>;
```
Comment (ChriLnth) [inline src/Provider.ts:44]: isnt account id a string?

**Review by ChriLnth — APPROVED** [review]
LGTM, just a comment

File: src/Provider.ts (hunk 1 above)
Comment (b0xtch) [reply src/Provider.ts:44]: I initially had it as string, then I saw this https://github.com/Psychedelic/plug-controller/blob/e579a7ba334af0ae73d855f24cb3afdba3a96b6e/src/PlugKeyRing/index.ts#L159
I could be missing some parsing

Comment (b0xtch) [reply src/Provider.ts:44]: `accoundId` refers to an index

**Pull request merged**